### PR TITLE
[WIP] Major Upgrading of Supabase Postgres

### LINKED ISSUE #35: Upgrade to postgres 13

### LINKED ISSUE #40: Compile postgres with LSE enabled
> So, if you’re running PostgreSQL on graviton2 instance, make sure you’ve binaries compiled with LSE support

http://akorotkov.github.io/blog/2021/04/30/arm/

currently the ansible role just [does apt-get](https://github.com/ANXS/postgresql/blob/e3f22bf58446039f24312583532280c7d9032e96/tasks/install_apt.yml#L17) and the [ubuntu binaries dont have LSE](https://github.com/aws/aws-graviton-getting-started#postgres) turned on..we might have to migrate away from the role to enable this which is a shame

### LINKED ISSUE #49: Install supautils in the image instead of in the worker
[Supautils](https://github.com/supabase/supautils) needs to be part of the image, so that we can track which version of supautils is installed in which postgres image

### PR DESCRIPTION
**Pull request by dragarcia**
## Primary Objectives
- [x] Closes #35 Bump to Postgres 13
- [x] Closes #40 Enable Large System Extensions
- [x] Closes #49 Install supautils as part of internal build 
- [x] System enhancements
    - [x] https://github.com/supabase/postgres/issues/36 Reconfigure fail2ban
- [x] Install popular requested extensions
    - [x] [timescaledb](https://github.com/supabase/supabase/discussions/679#discussioncomment-346352) 
    - [x] Select extensions from https://github.com/supabase/postgres/issues/45
- [ ] https://github.com/supabase/postgres/issues/46 Setup additional Marketplace images
    - [ ] PgBouncer Bundle
    - [ ] PostgRest Bundle
        - [x] Install PostgRest
        - [x] Make sure that it is not installed by default
    - [ ] Complete Bundle  

## Secondary/ Resultant Objectives
- [x] Shift away from [ANXS/postgresql](https://github.com/ANXS/postgresql)
    * Inevitable if we want to reap the benefits of LSE on Graviton 2 instances
    * It does however open up the following long-term benefits:
        * Increased granularity in modifying PostgreSQL given that we are now building it from source.
        * Able to immediately upgrade to the most up-to-date versions of Postgres
            * Before, we would need to wait for the above role to ensure that the newest version is available under them.
            * Now, we can upgrade as soon as the newest Postgres version is out
    - [x] Configure building Postgres from source
        - [x] Arm
        - [x] x86
    - [x] Add checks to verify Postgres installation
    - [x] Perform regression testing against other, existing tasks
        - [x] Arm
        - [x] x86
- [x] Ensure existing extensions are supported for Postgres 13
    - [x] Arm
    - [x] x86 
- [x] Re-establish versioning policy
    * After much thinking, I've changed my mind and realised that following the versioning policy of PostgreSQL could potentially simplify things.
        * `Major` would be dictated by Postgres
            * Would therefore only change whenever there is a new major of Postgres (eg. Postgres 13, Postgres 14, etc.) 
        * `Minor` would be dictated by Postgres
            * Would therefore only change whenever there is a new minor of Postgres (eg. Postgres 13.2, Postgres 13.3, etc.)
        *  `Patch` would be dictated by us
            * Would change whenever other updates other than the above are made to the image (eg. new extensions, system enhancements, etc.)
        * With this PR, the next version could potentially be `13.3.0`

### CODE REVIEW

**Comment by heliumbrain** [comment]
@dragarcia any specifics I could help out with here? Using your image for most of my Postgres-usage, would love to see a 13 version :) I could help you look in to the compatability of exisiting extension if you'd like

**Comment by dragarcia** [comment]
Hey @heliumbrain! 👋 

> I could help you look in to the compatability of exisiting extension if you'd like

I'm finishing up bumping the current versions of the existing extensions and testing them against Postgres 13. Should be pushing up the changes to this branch in the next few days. Would always be great to have another pair of eyes to see if everything is working 😄 . 

Just wondering, how are you currently using the Supabase Postgres image? Do you build it yourself with this repository, or do you get it from either the AWS/ DO marketplace?

**Comment by heliumbrain** [comment]
Cool @dragarcia!

Sounds great. Happy to have a look once that is done. 

I'm using the image as-is for a few projects and I'm building my own based on yours for some. Never tried it from one of the marketplaces

**Comment by dragarcia** [comment]
Alright, great! I'll update you again once I get these changes in so you can build the latest version yourself and try it out.

File: ansible/tasks/setup-supabase-internal.yml [hunk 1]
```diff
@@ -63,3 +70,48 @@
...
+
+- name: supautils - set supautils.reserved_memberships
+  become: yes
+  shell:
+    cmd: printf "supautils.reserved_memberships = 'pg_read_server_files, pg_write_server_files, pg_execute_server_program'" >> /etc/postgresql/postgresql.conf
+
+- name: supautils - set supautils.reserved_memberships
+  become: yes
+  shell:
+    cmd: printf "supautils.reserved_memberships = 'pg_read_server_files, pg_write_server_files, pg_execute_server_program'" >> /etc/postgresql/postgresql.conf
```
Comment (steve-chavez) [inline ansible/tasks/setup-supabase-internal.yml:112]: Seems there's a dup in these lines.

```suggestion
- name: supautils - set supautils.reserved_memberships
  become: yes
  shell:
    cmd: printf "supautils.reserved_memberships = 'pg_read_server_files, pg_write_server_files, pg_execute_server_program'" >> /etc/postgresql/postgresql.conf
```

Comment (dragarcia) [reply ansible/tasks/setup-supabase-internal.yml]: Removed the duplicated line. Thanks!

**Comment by dragarcia** [comment]
Hi @heliumbrain, this took longer to complete than expected. I'll be merging this into `develop` now. 

If you're still keen on building and trying out the image yourself (while we're in the midst of preparing the images for the AWS and Digital Ocean Marketplaces), you can always do so from the `develop` branch. Here are the steps to build your own image:
- via [AWS](https://github.com/supabase/postgres/wiki/AWS-EC2)
- via [Digital Ocean](https://github.com/supabase/postgres/wiki/Digital-Ocean)

**Pull request merged**